streams/src/test/java/org/apache/kafka/streams/state/internals/ReadonlyPartiallyDeserializedSegmentValueTest.java

```diff
@@ -0,0 +1,220 @@
+            for (final ResultOrder order: orders) {
+                for (final Long from : timestamps) {
+                    for (final Long to : timestamps) {
+                        // build expected results indices based on time range
+                        final List<Integer> expectedRecordsIndices = new ArrayList<>();
+                        for (int i = 0; i < testCase.records.size(); i++) {
+                            final long recordValidTo = i == 0 ? testCase.nextTimestamp : testCase.records.get(i - 1).timestamp;
+                            if (testCase.records.get(i).timestamp <= to && recordValidTo > from) {
+                                if (testCase.records.get(i).value != null) { // the results do not include tombstones
+                                    expectedRecordsIndices.add(i);
+                                }
+                            }
+                        }
+                        // The only object that has access to the find method is the instance of LogicalSegmentIterator.
+                        // Therefore, closing the iterator means that the segment calling the find method is destroyed and needs
+                        // to be created for the next time range.
+                        final ReadonlyPartiallyDeserializedSegmentValue segmentValue = buildSegmentWithInsertLatest(testCase);
+                        if (order.equals(ResultOrder.ASCENDING)) {
+                            Collections.reverse(expectedRecordsIndices);
+                        }
+                        int segmentRecordIndex = -1;
+                        for (final int index : expectedRecordsIndices) {
+                            final long expectedValidTo = index == 0 ? testCase.nextTimestamp : testCase.records.get(index - 1).timestamp;
+
+                            final SegmentSearchResult result = segmentValue.find(from, to, order, segmentRecordIndex);
+                            assert result != null;
+
+                            final TestRecord expectedRecord = testCase.records.get(index);
+
+                            assertThat(result.index(), equalTo(index));
+                            assertThat(result.value(), equalTo(expectedRecord.value));
+                            assertThat(result.validFrom(), equalTo(expectedRecord.timestamp));
+                            assertThat(result.validTo(), equalTo(expectedValidTo));
+
+                            segmentRecordIndex = order.equals(ResultOrder.ASCENDING) ? index - 1 : index + 1;
+                        }
+                        // verify no results within the time range
+                        if (expectedRecordsIndices.size() == 0) {
+                            assertNull(segmentValue.find(from, to, order, -1));
+                        }
+                    }
+                }
+            }
+        }
+
+        @Test
+        public void shouldGetTimestamps() {
```

Comment:
Seems we don't need this test? It just replicates an existing test, as it tests `RocksDBVersionedStoreSegmentValueFormatter`.